### PR TITLE
internal/v1: lookup user information by ID

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

File: candidclient/identity.go [hunk 1]
```diff
@@ -72,60 +73,55 @@ func (id *usernameIdentity) Domain() string {
...
-	}
-	return id.username, nil
+	id.user = *user
+	return string(id.user.Username), nil
 }
 
 // Groups implements Identity.Groups.
 func (id *useridIdentity) Groups() ([]string, error) {
-	username, err := id.Username()
+	_, err := id.Username()
```
Comment (tasdomas) [inline candidclient/identity.go:99]: Would it be more explicit if both `Groups()` and `Username()` called an internal function that would populate `id.user` ?

**Review by tasdomas — APPROVED** [review]
lgtm

**Review by urosj — APPROVED** [review]
LGTM

File: candidclient/identity.go (hunk 1 above)
Comment (mhilton) [reply candidclient/identity.go:99]: maybe, I don't think it would make it much simpler though.

**Pull request merged**